### PR TITLE
Reallocate natspec

### PR DESCRIPTION
**Pull request by QGarchery**
This PR:
- "documents" the changes done on https://github.com/morpho-org/metamorpho/pull/358
- clarifies that the order of operation matters in `reallocate`, because the extracted amount is used for future (supply) allocations

### CODE REVIEW

**Review by MerlinEgalite — APPROVED** [review]

File: src/interfaces/IMetaMorpho.sol [hunk 1]
```diff
@@ -164,8 +164,6 @@ interface IMetaMorphoBase {
     function updateWithdrawQueue(uint256[] calldata indexes) external;
 
     /// @notice Reallocates the vault's liquidity so as to reach a given allocation of assets on each given market.
-    /// @notice The allocator can withdraw from any market, even if it's not in the withdraw queue, as long as the loan
-    /// token of the market is the same as the vault's asset.
```
Comment (Rubilmax) [inline src/interfaces/IMetaMorpho.sol:168]: Why is this removed?

File: src/interfaces/IMetaMorpho.sol [hunk 2]
```diff
@@ -174,6 +172,8 @@ interface IMetaMorphoBase {
     /// - Withdrawals from markets that are expected to be withdrawn from during reallocation.
     /// @dev Sender is expected to pass `assets = type(uint256).max` with the last MarketAllocation of `allocations` to
     /// supply all the remaining withdrawn liquidity, which would ensure that `totalWithdrawn` = `totalSupplied`.
+    /// @dev A supply in a reallocation step will make the reallocation revert if the amount is greater than the amount
+    /// extracted (i.e. total withdrawn minus total supplied) in the previous steps.
```
Comment (Rubilmax) [inline src/interfaces/IMetaMorpho.sol:176]:
```suggestion
    /// extracted from the previous steps and remaining (i.e. total withdrawn minus total supplied).
```

File: src/interfaces/IMetaMorpho.sol [hunk 3]
```diff
@@ -174,6 +172,8 @@ interface IMetaMorphoBase {
     /// - Withdrawals from markets that are expected to be withdrawn from during reallocation.
     /// @dev Sender is expected to pass `assets = type(uint256).max` with the last MarketAllocation of `allocations` to
     /// supply all the remaining withdrawn liquidity, which would ensure that `totalWithdrawn` = `totalSupplied`.
+    /// @dev A supply in a reallocation step will make the reallocation revert if the amount is greater than the amount
```
Comment (adhusson) [inline src/interfaces/IMetaMorpho.sol:175]:
```suggestion
    /// @dev A supply in a reallocation step will make the reallocation revert if the amount is greater than the net amount
```

File: src/interfaces/IMetaMorpho.sol (hunk 2 above)
Comment (adhusson) [reply src/interfaces/IMetaMorpho.sol:176]: [alternative](https://github.com/morpho-org/metamorpho/pull/420/files/1cd6a0325b3691904056b446e2efa9d201e6daf7#r1633490943)

File: src/interfaces/IMetaMorpho.sol (hunk 1 above)
Comment (QGarchery) [reply src/interfaces/IMetaMorpho.sol:168]: Because it's false now, see the description of the PR (first point)

File: src/interfaces/IMetaMorpho.sol (hunk 2 above)
Comment (QGarchery) [reply src/interfaces/IMetaMorpho.sol]: I went with the alternative

**Review by MathisGD — APPROVED** [review]
